server/src/main/java/org/apache/druid/client/ServerView.java

```diff
@@ -29,7 +29,7 @@
  */
 public interface ServerView
 {
-  void registerServerRemovedCallback(Executor exec, ServerRemovedCallback callback);
+  void registerServerCallback(Executor exec, ServerCallback callback);
```

Comment:
Super nit:

To allow for more concise syntax (using lambdas) in the calling code, I wonder if we should have an overloaded method with the signature:

```java
  void registerServerCallback(Executor exec, Consumer<DruidServer> onAdded, Consumer<DruidServer> onRemoved);
```
assuming we get rid of the `CallbackAction` altogether.

`ServerCallback` can then be a concrete class which takes two `Consumer<DruidServer>` in its constructor, default values for which would be no-op.

It might be useful even when we add new methods to `ServerCallback` as callers would need to override only the required methods.

Reply:
A builder might be nice too. It would allow keeping together all the `ServerCallback` stuff in the interface. Something like:

```java
ServerCallback.builder()
              .onServerAdded(server -> ...)
              .onServerRemoved(server -> ...)
              .build()
```

or for just doing one:

```java
ServerCallback.onServerRemoved(server -> ...)
```

etc.

Would like to have this be a different PR, so that PR can focus on the thinking about the best way to specify callbacks that only want to listen to one thing.